<a href="https://colab.research.google.com/github/putrideafi-dev/latihanEDAdeafi/blob/main/Latihan_eda_numpy_pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np

harga = np.array([5000, 7000, 3000, 12000, 4500])

print('Rata-rata harga:', harga.mean())
print('Harga tertinggi:', harga.max())
print('Harga setelah diskon 10%:', harga * 0.9)

Rata-rata harga: 6300.0
Harga tertinggi: 12000
Harga setelah diskon 10%: [ 4500.  6300.  2700. 10800.  4050.]


In [2]:
import pandas as pd

data_kantin = {
    'menu': ['Nasi Goreng', 'Es Teh', 'Mie Ayam', 'Es Teh', None],
    'harga': [12000, 4000, 10000, 4000, 8000],
    'terjual': [23, 40, None, 35, 18]
}

df = pd.DataFrame(data_kantin)
print(df)

          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4         None   8000     18.0


In [4]:
df = pd.read_csv('data_kantin.csv')

In [5]:
df = pd.read_csv('data_kantin.csv')

print(df.head())
print(df.info())
print(df.describe())
print(df.shape)

      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB
None
              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.000000 

In [6]:
print(df.isnull().sum())

df['terjual'] = df['terjual'].fillna(0)

df = df.dropna(subset=['menu'])

print(df)

tanggal     0
menu        3
kategori    0
harga       0
terjual     6
dtype: int64
        tanggal         menu kategori  harga  terjual
0    2026-08-20   Teh Hangat  Minuman   3000     11.0
1    2026-08-31  Nasi Goreng  Makanan  12000     29.0
2    2026-08-21  Nasi Goreng  Makanan  12000     28.0
3    2026-08-28   Teh Hangat  Minuman   3000     12.0
4    2026-08-21     Es Jeruk  Minuman   5000     19.0
..          ...          ...      ...    ...      ...
105  2026-09-01     Es Jeruk  Minuman   5000     24.0
106  2026-09-04  Nasi Goreng  Makanan  12000     28.0
107  2026-08-19      Kerupuk    Snack   2000     37.0
108  2026-08-25       Es Teh  Minuman   4000     44.0
109  2026-09-02      Kerupuk    Snack   2000     34.0

[107 rows x 5 columns]


In [7]:
print(df.duplicated().sum())

df = df.drop_duplicates()

df['harga'] = df['harga'].astype(int)

print(df.dtypes)

5
tanggal      object
menu         object
kategori     object
harga         int64
terjual     float64
dtype: object


In [8]:
laris = df[df['terjual'] > 20]

urut = df.sort_values(by='terjual', ascending=False)

df['total_pendapatan'] = df['harga'] * df['terjual']

ringkasan = df.groupby('menu')['total_pendapatan'].sum()

print("Menu yang terjual lebih dari 20:")
print(laris)

print("\nUrutan berdasarkan jumlah terjual:")
print(urut)

print("\nTotal pendapatan setiap menu:")
print(ringkasan)

Menu yang terjual lebih dari 20:
        tanggal         menu kategori  harga  terjual
1    2026-08-31  Nasi Goreng  Makanan  12000     29.0
2    2026-08-21  Nasi Goreng  Makanan  12000     28.0
5    2026-08-27     Es Jeruk  Minuman   5000     21.0
8    2026-08-25      Kerupuk    Snack   2000     32.0
12   2026-08-17  Nasi Goreng  Makanan  12000     23.0
16   2026-09-01       Es Teh  Minuman   4000     45.0
17   2026-09-02     Es Jeruk  Minuman   5000     21.0
22   2026-08-24     Es Jeruk  Minuman   5000     23.0
26   2026-08-19       Es Teh  Minuman   4000     45.0
31   2026-08-17      Kerupuk    Snack   2000     35.0
32   2026-08-28  Nasi Goreng  Makanan  12000     27.0
34   2026-08-31       Es Teh  Minuman   4000     38.0
38   2026-08-24  Nasi Goreng  Makanan  12000     22.0
39   2026-08-27       Es Teh  Minuman   4000     43.0
44   2026-09-04      Kerupuk    Snack   2000     36.0
46   2026-08-26     Es Jeruk  Minuman   5000     22.0
47   2026-08-26  Nasi Goreng  Makanan  12000     

Jawaban Tugas Analisis
Latihan 1 : NumPy array dapat dikalikan dengan 0.9 untuk semua elemen sekaligus. Sedangkan list Python [5000,7000,3000] * 0.9 akan menghasilkan error karena list tidak mendukung perkalian dengan desimal.

Jawaban Tugas Analisis
Latihan 2 : Data kosong terdapat pada kolom menu dan terjual. Jika tidak dibersihkan, data kosong dapat menyebabkan hasil analisis menjadi tidak akurat.

Jawaban Tugas Analisis 3 : Kolom yang jumlah non-null-nya lebih sedikit adalah menu (107) dan terjual (104). Artinya, terdapat 3 data kosong pada menu dan 6 data kosong pada terjual.

Jawaban Tugas Analisis 4 : Kolom terjual diisi dengan 0 karena data kosong dapat dianggap tidak ada penjualan. Sedangkan baris dengan menu kosong dihapus karena nama menu penting untuk analisis, sehingga data tanpa nama menu tidak dapat digunakan dengan baik.

Jawaban Tugas Analisis 5 : terdapat 5 data duplikat yang dihapus, sehingga jumlah baris berkurang 5 dari sebelumnya. Memastikan tipe data benar penting agar perhitungan dan analisis data dapat dilakukan dengan tepat.

Jawaban Tugas Analisis 6 : Menu dengan total_pendapatan tertinggi adalah Nasi Goreng, yaitu Rp3.924.000. Informasi ini dapat membantu kantin menentukan menu yang perlu lebih banyak disediakan dan menjadi pertimbangan dalam strategi penjualan.